builder/azure/common/artifact.go

```diff
@@ -54,10 +58,7 @@ func (a *Artifact) String() string {
 }
 
 func (a *Artifact) State(name string) interface{} {
-	switch name {
-	default:
-		return nil
-	}
+	return a.StateData[name]
```

Comment:
Is it possible for StateData to ever be nil? If so, this might fail.

Reply:
Right now is not possible because I'm setting this value inside the builder's run method but I think it's a good idea to make it sure to check this before using it! I will change this :D

Reply:
Actually I've just found out that if StateData is nil this one doesn't fail and will only return nil as expected. I added a test for this for each of the artifacts.